src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs

```diff
@@ -155,6 +155,22 @@ internal sealed class ManagedDecryptorPal : DecryptorPal
                         }
                     }
                 }
+                else
+                {
+                    using (AsnWriter writer = new AsnWriter(AsnEncodingRules.DER))
+                    {
+                        writer.PushSequence();
+                        AsnReader reader = new AsnReader(decrypted, AsnEncodingRules.DER);
```

Comment:
Most of CMS says that it's BER, meaning indefinite length encodings are permitted.  This may be overly restrictive.  (Same with the writer, or it will validate the call to WriteEncodedValue)

Reply:
@bartonjs I did it this way because Windows appends DER prefix and I wasn't sure how to enforce length definite Sequence tag. I was thinking about octet string instead of this and replacing the first byte to `0x30` to avoid validation - I haven't checked what BER produces by default though so I presume you're right about being to restrictive (I also thought this is always DER).